### ISSUE TITLE
NSRegularExpression Nil Argument Crash

### ISSUE THREAD

**Issue by wimbledon** [pubnative/pubnative-hybid-ios-sdk#321]
@can-soykarafakili, I just released my app to AppStore with HyBid iOS SDK 2.4.0 and I'm getting the following crash in production. Please advise mitigation strategy and fix as soon as possible.

**Fatal Exception: NSInvalidArgumentException
*** -[NSRegularExpression enumerateMatchesInString:options:range:usingBlock:]: nil argument**
_Occurs from iOS 12 - iOS 14 in the foreground affecting ~23% of total users._
```
Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x1822ed86c __exceptionPreprocess
1  libobjc.A.dylib                0x197308c50 objc_exception_throw
2  Foundation                     0x1835527bc -[NSRegularExpression numberOfCaptureGroups]
3  Foundation                     0x183572998 -[NSRegularExpression(NSMatching) matchesInString:options:range:]
4  Foundation                     0x18357ac60 -[NSRegularExpression(NSReplacement) stringByReplacingMatchesInString:options:range:withTemplate:]
5  HyBid                          0x1031981e4 +[PNLiteMRAIDUtil processRawHtml:] + 43 (PNLiteMRAIDUtil.m:43)
6  HyBid                          0x1031659c0 -[HyBidMRAIDView initWithFrame:withHtmlData:withBaseURL:asInterstitial:supportedFeatures:delegate:serviceDelegate:rootViewController:contentInfo:] + 246 (HyBidMRAIDView.m:246)
7  HyBid                          0x1031653ec -[HyBidMRAIDView initWithFrame:withHtmlData:withBaseURL:supportedFeatures:isInterstital:delegate:serviceDelegate:rootViewController:contentInfo:] + 166 (HyBidMRAIDView.m:166)
8  HyBid                          0x103193b14 -[PNLiteMRAIDBannerPresenter load] + 60 (PNLiteMRAIDBannerPresenter.m:60)
9  HyBid                          0x10318cc34 -[PNLiteAdPresenterDecorator load] + 45 (PNLiteAdPresenterDecorator.m:45)
10 HyBid                          0x10315d68c -[HyBidAdView renderAd] + 175 (HyBidAdView.m:175)
11 HyBid                          0x10315dd4c -[HyBidAdView request:didLoadWithAd:] + 228 (HyBidAdView.m:228)
12 HyBid                          0x103158f1c __32-[HyBidAdRequest invokeDidLoad:]_block_invoke + 189 (HyBidAdRequest.m:189)
13 libdispatch.dylib              0x181ee024c _dispatch_call_block_and_release
14 libdispatch.dylib              0x181ee1db0 _dispatch_client_callout
15 libdispatch.dylib              0x181eef7ac _dispatch_main_queue_callback_4CF
16 CoreFoundation                 0x18226911c __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
17 CoreFoundation                 0x182263120 __CFRunLoopRun
18 CoreFoundation                 0x18226221c CFRunLoopRunSpecific
19 GraphicsServices               0x199e2e784 GSEventRunModal
20 UIKitCore                      0x184ca2ee8 -[UIApplication _run]
21 UIKitCore                      0x184ca875c UIApplicationMain
```

**Comment by wimbledon** [comment]
@eros902002, do you want to close this bug as it is resolved in SDK 2.4.1?

**Comment by eros902002** [comment]
@wimbledon Yes we will close it as solved. Just waiting to finish the pod release and we'll do it 😄

**Comment by eros902002** [comment]
This issue has been solved on the release 2.4.1
